src/hssm/distribution_utils/dist.py

```diff
@@ -356,8 +357,210 @@ def rng_fn(
+
+            iinfo32 = np.iinfo(np.uint32)
+            seed = rng.integers(0, iinfo32.max, dtype=np.uint32)
+
+            is_all_args_scalar = all(arg.size == 1 for arg in arg_arrays)
+
+            if is_all_args_scalar:
+                # All parameters are scalars
+
+                theta = np.stack(arg_arrays)
+                if theta.ndim > 1:
+                    theta = theta.squeeze(axis=-1)
+
+                theta = np.tile(theta, (size, 1))
+            else:
+                # Preprocess all parameters, reshape them into a matrix of dimension
+                # (size, n_params) where size is the number of elements in the largest
+                # of all parameters passed to *arg
+
+                elem_max_size = np.argmax([arg.size for arg in arg_arrays])
+                max_shape = arg_arrays[elem_max_size].shape
+
+                theta = np.column_stack(
+                    [np.broadcast_to(arg, max_shape).reshape(-1) for arg in arg_arrays]
+                )
+
+            sims_out = simulator_fun(
+                theta=theta,
+                random_state=seed,
+                **kwargs,
+            )
+
+            if p_outlier is not None:
+                assert cls._lapse is not None, (
+                    "You have specified `p_outlier`, the probability of the lapse "
+                    + "distribution but did not specify the distribution."
+                )
+                out_shape = sims_out.shape[:-1]
+                if not np.isscalar(p_outlier) and len(p_outlier.shape) > 0:
+                    if p_outlier.shape[-1] == 1:
+                        p_outlier = np.broadcast_to(p_outlier, out_shape)
+                    else:
+                        p_outlier = p_outlier.reshape(out_shape)
+
+                replace = rng.binomial(n=1, p=p_outlier, size=out_shape).astype(bool)
+                replace_n = int(np.sum(replace, axis=None))
+                if replace_n == 0:
+                    return sims_out
+
+                replace_shape = (*out_shape[:-1], replace_n)
+                replace_mask = np.stack([replace, replace], axis=-1)
+                n_draws = np.prod(replace_shape)
+                lapse_rt = pm.draw(
+                    get_distribution_from_prior(cls._lapse).dist(**cls._lapse.args),
+                    n_draws,
+                    random_seed=rng,
+                ).reshape(replace_shape)
+
+                lapse_response = rng.choice(
+                    choices,
+                    p=1 / len(choices) * np.ones(len(choices)),
+                    size=replace_shape,
+                )
+                lapse_output = np.stack(
+                    [lapse_rt, lapse_response],
+                    axis=-1,
+                )
+                np.putmask(sims_out, replace_mask, lapse_output)
+            return sims_out
+
+    return CustomRV
+
+
 def make_distribution(
-    rv: str | Type[RandomVariable],
+    rv: str | Type[RandomVariable] | RandomVariable | Callable,
```

Comment:
This argument can be confusing now since it also accepts a callable. Might be a good idea to have a separate `simulator_fn` argument if someone wants to pass a simulator